### ISSUE TITLE
Comma-separated article stati for <txp_article />

### ISSUE THREAD

**Issue by ghost** [textpattern/textpattern#177]
_From [r.wetzlmayr](https://code.google.com/u/r.wetzlmayr/) on April 12, 2012 07:19:08_

<txp_article /> might allow a comma separate list of vlaues for the 'status' attribute: <txp:article status="live,sticky" />

To clarify: 
- what determines article list order
- how would paging work
- ...

_Original issue: http://code.google.com/p/textpattern/issues/detail?id=173_

**Comment by petecooper** [comment]
+1 for this issue, I have 2 (small) clients asking if it's viable.

**Comment by bloatware** [comment]
Currently, `status="live"` and `status="sticky"` act differently re search, pagination and frontpage filtering. For example, `sticky` ignores `searchform` attribute. So, which behavior should be retained for mixed lists? I guess we switch to `live` mode as soon as `status` contains `live`? Any drawbacks?

**Comment by petecooper** [comment]
In layman's terms, a `sticky` article is visible from the front-side, and could feasibly be considered `live` _and_ `sticky`. Does that extend to behind-the-scenes logic, too?

**Comment by Bloke** [comment]
This has been batted around since (I think) 4.0.6 or maybe earlier when the feature was dropped in favour of only permitting one status. There were legitimate reasons at the time for only considering one status because it leads to ambiguous or unexpected output, as Oleg mentions.

The question is: has the Txp conceptual model or (internally) the way article(_custom) is handled changed sufficiently to make this a viable reintroduction? That's a speaking-out-loud question, btw, as I don't know.

This has some basis in my (failed) attempts in the status-mods branch. The basic premise I was trying to address in that branch was that, instead of making the assumption that stati 1, 2, 3 are "unpublishable" and anything >= 4 is publishable, to explicitly use each. For example:

`if (status >= STATUS_LIVE)  {...}`

effectively becomes:

`if (status === STATUS_LIVE || status === STATUS_STICKY) {...}`

It's abit more sublte than that (it uses groups). And I would have got away with it if it hadn't been for you pesky `getNeighbour()` functionality. See [this monologue](https://forum.textpattern.io/viewtopic.php?id=47407) for details.

If things have moved on since then (I know `getNeighbour()` has changed a lot) then maybe we can resusicitate this idea in the next release and imrpove things not only for the tags, but enable plugins to make custom workflows too?

**Comment by petecooper** [comment]
Thanks @Bloke and @bloatware - insightful as always.

Proposal: throw a `4.8` milestone on it and kick it to a back burner.

**Comment by bloatware** [comment]
FWIW, `getNeighbour()` has mainly changed to fix navigation when articles are sorted by some non-unique field, e.g. `Section`.

**Comment by Bloke** [comment]
I was wondering/hoping if the changes to support non-unique fioelds might also extend to multiple statuses, which are non-unique by their nature :-)

**Comment by bloatware** [comment]
You can sort by `Status`, no worries :-)

**Comment by petecooper** [comment]
Bumping this – it's drifted a little since OP, but is it addressed by https://github.com/textpattern/textpattern/commit/f9eb1ac655903e66a68914f8215fed17ea326802 ?

**Comment by petecooper** [comment]
I'll close this since the desired output can be achieved, but not strictly within the bounds of OP's syntax.
